intake/catalog/tests/test_local.py

```diff
@@ -79,6 +79,13 @@ def test_nested(catalog1):
     assert catalog1.nested.nested.nested().cat.cat.cat is catalog1
 
 
+def test_nested_gets_name_from_super(catalog1):
+    assert catalog1.name == 'name_in_cat'
+    assert 'nested' in catalog1
+    nested = catalog1.nested
+    assert nested.name == 'nested'
```

Comment:
Shouldn't the name in the spec trump the name of the entry? What is `nested().name` ?

Reply:
I guess that is what I have written at the top of this PR, but it wasn't clear to me which should win out of 2 or 3. Is it weird to not have the name match the key?

Reply:
I think entry name and source name need to be the same: `nested().name == 'nested'`

Reply:
Do you have a preference?

Reply:
What about this one?

Reply:
I would go with what you suggest. Perhaps the docstring for YAMLFile(s)Catalog can say that the cat object gets a generated name *if* it wasn't already supplied.